service/migration_manager.cc

```diff
@@ -516,6 +517,29 @@ future<> migration_notifier::update_view(view_ptr view, bool columns_changed) {
     });
 }
 
+static future<service::storage_service*> get_storage_service(atomic_vector<migration_listener*>& listeners) {
+    service::storage_service* ssl = nullptr;
+    co_await listeners.for_each([&] (migration_listener* l) -> future<> {
+        auto ss = dynamic_cast<service::storage_service*>(l);
+        if (ss) {
+            ssl = ss;
+        }
+        co_return;
+    });
+    SCYLLA_ASSERT(ssl);
+    co_return ssl;
+}
```

Comment:
Pass storage_service via the constructor, not this backhanded way.

Reply:
Or, make storage_service a migration_listener (is it not already?!)

Reply:
added dependency, it's not possible to use migration_listener because one of the calls returns a results which is passed to another (which multiple subscribers there is a matching problem, and it's weird notification handling anyway)